### PR TITLE
Fix theme flashing

### LINKED ISSUE #134: Default Styling set to Dark Mode
This is an issue because if you select light mode as your preferred theme, some pages will start with the dark mode color scheme and then change to the light mode color scheme. It can look pretty jarring.

You can argue that my implementation might not have been the best but I think there's a more general issue of how to serve the pages to the user only after the settings have been applied to the page. Solving this issue would solve the bug here.

### PR DESCRIPTION
**Pull request by paytonshaltis**
Closes #134.

Settings are now preloaded into the HTML page that is sent to the client, meaning you will no longer get an odd flash if your theme is light (different from the default dark). This was done by putting much of the asynchronous settings logic from the `main.js` files into a partial that is loaded when the HTML is being assembled on the server side.

The flash that a user gets when loading their home page or profiles page can be fixed in a similar manner, so if we think @sderacy @lelekaz @matyis1 @thadakn1 that this small flashing is problematic, let me know and I'll create a new issue for it.

The EJS logic in the partials is marked as an error by VSCode, and the only workaround is to change your scrtipt validation in HTML files [as discussed here](https://stackoverflow.com/questions/66827697/writing-ejs-in-a-script-is-giving-me-the-error-expression-expected). Note that there is not an error with the code, but VSCode just doesn't play well with the syntax of EJS within HTML.

You'll notice that if you refresh fast enough, there is sometimes still an issue of some minor flashing. From what I have seen, this may just be a caching thing and happens on other sites too.

For the future, if something needs to be fetched asynchronously from the database and it is done in the main.js file, it will always be done with a slight delay because the sent file does not execute the JS until the client gets it. The whole point of EJS is to assemble the HTML with default values that may need to be fetched from the database on the _server-side_, then send over the assembled page without needing to do a fetch after the fact.

### CODE REVIEW

File: views/aboutus/main.js [hunk 1]
```diff
@@ -1,58 +0,0 @@
-// Make sure the settings are fetched.
```
Comment (sderacy) [inline views/aboutus/main.js:1]: So if you look at the about us page, it is still currently being set to white even in light mode. Where exactly is this being stored?

File: views/partials/defaultSettings.ejs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<!-- Apply the settings using the passed EJS user object. -->
```
Comment (sderacy) [inline views/partials/defaultSettings.ejs:1]: I think I see what's happening here. So the only issue with this approach is that not every page follows the exact same format so some pages (like on aboutus) needed more properties to be modified than others

**Comment by sderacy** [comment]
> The flash that a user gets when loading their home page or profiles page can be fixed in a similar manner, so if we think @sderacy @lelekaz @matyis1 @thadakn1 that this small flashing is problematic, let me know and I'll create a new issue for it.

I can live with it.

**Comment by sderacy** [comment]
Fixed it, looks good otherwise

**Review by sderacy — APPROVED** [review]
LGTM

File: views/partials/defaultSettings.ejs (hunk 2 above)
Comment (paytonshaltis) [reply views/partials/defaultSettings.ejs:1]: > I think I see what's happening here. So the only issue with this approach is that not every page follows the exact same format so some pages (like on aboutus) needed more properties to be modified than others

Exactly. Specifically, the settings page has a lot of things that need to be done before being sent to the user. Besides just applying the settings to the page, the form needs to have its default values set based on the user's settings selection, which is why the other partial is just used for this page.

**Pull request merged**